src/spikeinterface/postprocessing/template_metrics.py

```diff
@@ -141,8 +150,24 @@ def _set_params(
             metrics_kwargs_ = _default_function_kwargs.copy()
             metrics_kwargs_.update(metrics_kwargs)
 
+        tm_extension = self.sorting_analyzer.get_extension("template_metrics")
+        if delete_existing_metrics is False and tm_extension is not None:
+
+            existing_params = tm_extension.params["metrics_kwargs"]
+            # checks that existing metrics were calculated using the same params
+            if existing_params != metrics_kwargs_:
+                warnings.warn(
+                    "The parameters used to calculate the previous template metrics are different than those used now. Deleting previous template metrics..."
+                )
+                tm_extension.params["metric_names"] = []
+                existing_metric_names = []
+            else:
+                existing_metric_names = tm_extension.params["metric_names"]
+
+            metric_names = list(set(existing_metric_names + metric_names))
```

Comment:
The set is making a reordering.
Maybe this could be usefull for a user to keep the same order of metric_names inside the params no ?

Reply:
Yes - the automated curation stuff uses metrics and ordering is important. So let's keep any user-specified order. Done!